src/main/xar-resources/data/urlrewrite/urlrewrite.xml

```diff
@@ -10,175 +8,169 @@
-            returns that prefix.</para>
-          <para>
-            For example: the default configuration maps the path
-            <literal>/tools</literal>
-            to a collection in the database (see below). In this case,
-            <literal>$exist:prefix</literal>
-            would contain
-            <literal>/tools</literal>.</para>
+          <para><code>XQueryURLRewrite</code> uses the controller XML fragment as an instruction on what to do next.  This instruction may be as simple as forwarding to a resource on (or off) the server, or it may be as complex as a pipeline using the <emphasis>Model-View-Controller</emphasis> pattern (see <xref linkend="mvc-pipelines"/>) and other servlets such as eXist's <xref linkend="xq-servlet"/> or <xref linkend="xslt-servlet"/>.</para>
         </listitem>
-      </varlistentry>
-      <varlistentry>
-        <term>
-          <code>exist:root</code>
-        </term>
+      </orderedlist>
+    <sect2 xml:id="eg1">
+      <title>Example I: A Simple Implementation</title>
+      <para>Consider  the document you are currently reading; a direct URL pointing to the source data might be <code>/exist/apps/doc/data/urlrewrite.xml</code>.  But accessing this URI  would only show a user the raw XML: instead users should get a properly formatted HTML version of the text, accessible through a simple URL like <code>/exist/apps/doc/urlrewrite</code>.</para>
+      <para>To achieve this we need a mechanism to map or rewrite a given URL to an application specific endpoint. So in the simplest case, calling <code>/exist/apps/doc/urlrewrite</code> would locate the source document, transform it into HTML and return the result.</para>
+      <para><literal>controller.xq</literal> is invoked for all URL paths targeting the collection in which it resides. It has access to a number of <xref linkend="variables"/> pre-filled with details about the request, including <literal>$exist:resource</literal>, containing the name of the resource (without path components) the request tries to access; also the <literal>$exist:controller</literal> variable which points to the collection the <literal>controller.xq</literal> is located in.</para>
+      <para>For example, one may want to direct all requests to <literal>/exist/apps/doc/{resource}</literal> to an XQuery, <literal>transform.xq</literal>, which is responsible for converting the XML content into HTML:</para>
+      <programlisting language="xquery" xlink:href="listings/listing-1.txt"/>
+      <para>This example controller returns a simple <tag>dispatch</tag> fragment which will be passed back to the URL rewriting framework. The <tag>forward</tag> element instructs the framework to call the URL <literal>modules/transform.xq</literal> relative to the collection in which the controller resides. It adds a request parameter, named <literal>doc</literal>, which translates the requested resource into an actual document path to be transformed.</para>
```

Comment:
I'm not sure it's totally accurate to say that the request parameter named `doc` "translates the requested resource into an actual document path". Instead, `doc` contains the name of the resource sans path components (e.g., `urlrewrite`) plus the file extension of the actual document stored in the database (e.g., `urlrewrite.xml`).